src/accelerate/utils/imports.py

```diff
@@ -444,3 +444,9 @@ def is_torchdata_stateful_dataloader_available():
         torchdata_version = version.parse(importlib.metadata.version("torchdata"))
         return compare_versions(torchdata_version, ">=", "0.8.0")
     return False
+
+
+def is_weights_only_available():
+    # Weights only with allowlist was added in 2.4.0
+    # ref: https://github.com/pytorch/pytorch/pull/124331
+    return is_torch_version(">=", "2.4.0")
```

Comment:
The other option is to ignore the allowlist and force `weights_only` from v2.0.0, WDYT @BenjaminBossan.

(Basically this will wind up allowing only the optimizer, model, and scheduler)

Reply:
The way that we decided to solve this in torch is to stay with the current default in torch, which is `weights_only=None` (despite the docs saying otherwise), which gives the warning and then switches to `weights_only=False`. For torch 2.6 (which is projected to make the switch), we also switch to `weights_only=True`.

The reasoning was that we stick with what torch does and once torch makes the switch, users would run into potential errors no matter what.

If you feel confident that this won't disrupt accelerate users, or if it does it's better to do it earlier rather than later, it's also fine to switch now. I just think that when the corresponding torch version releases, a lot of code will break and there will be general chatter and awareness of the problem, making it easier for users to find help for their breaking code.

Reply:
If I ran into any sort of flag/issue with the checkpointing script, I'd have been more likely to not. Just in case I'll run the slow suite, but if nothing breaks I think we're safe here.

Reply:
Makes sense.